tests/subsys/spm/secure_services/src/main.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <ztest.h>
+#include <secure_services.h>
+#include <pm_config.h>
+
+const struct fw_info ns_fw_info = { .magic = { FIRMWARE_INFO_MAGIC } };
+
+void test_spm_firmware_info(void)
+{
+	int err;
+	struct fw_info info;
+
+	/* Prove that the struct is stored in non-secure flash. */
+	zassert_true((uint32_t)&ns_fw_info > PM_ADDRESS,
+		     "Test fw-info is not stored in non-secure firmware");
+
+	/* Normal execution */
+	err = spm_firmware_info(PM_SPM_ADDRESS, &info);
+	zassert_equal(err, 0, "Got unexpected failure");
+
+	/* Verify that the function fails if the address of the firmware info
+	 * is not in secure firmware.
+	 */
+	err = spm_firmware_info((uint32_t)&ns_fw_info, &info);
+	zassert_true(err != 0, "Did not fail for fw_address in secure fw");
+
+	/* Verify that the function fails if the output argument for the info
+	 * is in secure RAM. First verify that it passes when it is in
+	 * non-secure, indicating that a valid fw_info is found at the
+	 * specified address, then verify that it fails for an output pointer
+	 * in secure RAM.
+	 */
+	err = spm_firmware_info(PM_SPM_ADDRESS,
+				(struct fw_info *)PM_SPM_SRAM_ADDRESS);
+	zassert_true(err != 0, "Did not fail when passing invalid output ptr");
+}
+
```

Comment:
Can you do a quick test for the other secure services as well, that they fail for pointers in secure RAM?

Reply:
Absolutely, good idea!

Reply:
added tests for all but s0_active, but that is just a bool pointer, so I guess you cannot do that much harm. Will see if I can add a test for that as well.

Reply:
adding s0_active now as well.